### PR TITLE
Rework Database

### PR DESCRIPTION
**Pull request by dholms**
- remove sqlite support & replace it with a mock DB for dev environments
- integrate bryan's feedback on PG schemas
  - JSONB type for operation
  - boolean & no default for nullified 
  - timestamptz type for createdAt with default to current_timestamp
  - index on createdAt

### CODE REVIEW

**Comment by bnewbold** [comment]
The server pg test setup isn't working for me, probably a docker/docker-compose version issue. It seems to be working in CI, so probably an issue with my setup that I will debug.

File: packages/server/src/db/index.ts [hunk 1]
```diff
@@ -165,58 +173,49 @@ export class Database {
...
+    //     .execute()
+    //   return res.reduce((acc, cur) => {
+    //     acc[cur.did] ??= []
+    //     acc[cur.did].push({
+    //       op: cur.operation),
+    //       nullified: cur.nullified === 1,
+    //       createdAt: cur.createdAt,
+    //     })
+    //     return acc
+    //   }, {} as Record<string, OpLogExport>)
```
Comment (bnewbold) [inline packages/server/src/db/index.ts:205]: I assume this is TODO work for the later export PR

File: packages/server/src/migrations/20230223T215019669Z-refactor.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+export async function up(db: Kysely<any>): Promise<void> {
+  await db.schema
+    .createTable('operations_new')
+    .addColumn('did', 'text', (col) => col.notNull())
+    .addColumn('operation', 'jsonb', (col) => col.notNull())
+    .addColumn('cid', 'text', (col) => col.notNull())
+    .addColumn('nullified', 'boolean', (col) => col.notNull())
+    .addColumn('createdAt', 'timestamptz', (col) =>
+      col.defaultTo(sql`current_timestamp`),
+    )
```
Comment (bnewbold) [inline packages/server/src/migrations/20230223T215019669Z-refactor.ts:12]: Even with the `DEFAULT`, this should be `NOT NULL` (unless Kysely is doing that automatically).

File: packages/server/tests/_util.ts [hunk 3]
```diff
@@ -6,23 +6,23 @@ export type CloseFn = () => Promise<void>
...
-  const dbPostgresUrl = process.env.DB_POSTGRES_URL || undefined
-
-  const db =
-    dbPostgresUrl !== undefined
-      ? Database.postgres({
-          url: dbPostgresUrl,
-          schema: dbPostgresSchema,
-        })
-      : Database.memory()
+  const dbPostgresUrl = process.env.DB_POSTGRES_URL
```
Comment (bnewbold) [inline packages/server/tests/_util.ts:17]: FWIW, in the golang code we are using `DATABASE_URL` with `postgresql://` prefix for postgresql backend (or `sqlite://`, though you don't do sqlite in this project any more). This is a sort-of-standard way to do it (choice of variable name).

But i'm not sure it is best to do that here b/c you are sticking with existing behavior. And this is just a soft norm, not any real win from one variable name or another.

**Review by bnewbold — COMMENTED** [review]
Gave it a quick look, mostly the SQL schema. Have a note about the `createdAt` column being `NOT NULL`.

I don't know enough about Kysely migrations as to whether it is safe to run the migration over an existing database which already has a different set of Kysely migration tabels initialized. I would have assumed that the `20221020T204908820Z-o
perations-init.ts` file would have needed an "IF NOT EXISTS" or something, but maybe it is picking up the existing state magically. The CI tests run so I guess it works!

File: packages/server/src/migrations/20230223T215019669Z-refactor.ts (hunk 2 above)
Comment (dholms) [reply packages/server/src/migrations/20230223T215019669Z-refactor.ts:12]: ah you're right 👍

File: packages/server/tests/_util.ts (hunk 3 above)
Comment (dholms) [reply packages/server/tests/_util.ts:17]: Don't see a huge different here, but I'm down to change it

Bigger different in the PDS code, but we can keep all our impl in sync

**Review by bnewbold — APPROVED** [review]
With changes, looks :100:

**Pull request merged**